src/org/traccar/FilterHandler.java

```diff
@@ -150,16 +150,40 @@ private boolean filterLimit(Position position) {
 
     private boolean filter(Position p) {
```

Comment:
If you break down filters into separate ifs, please rename the variable to "position".